Edits to JOSS paper

kyleniemeyer opened the pull request:
Edits to the JOSS paper for the Thermal Hydraulics Module: https://github.com/openjournals/joss-reviews/issues/6146

kyleniemeyer committed:
Edits to JOSS paper

kyleniemeyer committed:
JOSS references fixes

Adds some missing DOIs, and fixes capitalization

kyleniemeyer committed:
more JOSS references updates

some missing initials and author names

kyleniemeyer committed:
Replaces website links in JOSS website

Replaces website citation with direct links

joshuahansel commented on joss_paper.bib:
```diff
@@ -14,16 +14,15 @@ @article{lindsay2022moose
 author = {R. A. Berry and L. Zou and H. Zhao and H. Zhang and J. W. Peterson and R. C. Martineau and S. Y. Kadioglu and D. Andrs},
 institution = {Idaho National Laboratory},
-number = {INL/EXT-14-31366},
+howpublished = {INL/EXT-14-31366},
```
I think `number` is the appropriate field - why change?

kyleniemeyer replied:
ah, I thought that our citation formatting didn't recognize `number` for `techreport` type, but at second glance I can see that it did show up property. More importantly, we should add the DOI 10.2172/1262488

joshuahansel commented on joss_paper.bib:
```diff
@@ -65,21 +64,24 @@ @article{slaughter2021continuous
   institution={Idaho National Laboratory},
   year={2020},
-  note = {Revision 7}
+  note = {Revision 7},
```
I don't think a comma should go after the last item, but maybe this is a moot point if this reference needs to be removed anyway (I'm currently asking around if there's a public SQA plan reference).

kyleniemeyer replied:
This is just force of habit on my part—commas after the final list entry are totally acceptable, but yes unnecessary. (Same is true in most programming language lists, dicts, etc.).

kyleniemeyer committed:
Updated RELAP-7 theory citation

refs #22309

kyleniemeyer commented:
@joshuahansel I updated the RELAP-7 theory manual citation, to use `number` again and also updated to the 2018 version, with appropriate DOI: https://www.osti.gov/biblio/1498781

kyleniemeyer commented:
Thanks @GiudGiud - I just amended the last commit to include that reference and force-pushed it.

joshuahansel commented:
> Looks like we wont allow civet to run on PRs to branches that are not next @joshuahansel we may want to change that

I think this is fine, just a little awkward: this PR into my feature branch is untested (fine), and then I'll do a PR into next from this branch, which you can review and is tested.

joshuahansel commented:
@kyleniemeyer I checked on the SQA reference. There is no public reference for the plan itself, but the first reference in that section (slaughter2021continuous) does discuss the plan a bit, so I propose that we delete the reference and the corresponding entry in the .bib file. If you approve, then either you can apply those changes, or we merge this one and then I do it afterwards.

kyleniemeyer committed:
Replaces MOOSE SQA reference

Replaces reference to the private/internal SQA document with a published article
that discusses the SQA plan. Also updates that reference with final bib data.

refs #22309

kyleniemeyer commented:
@joshuahansel ok, I went ahead and replaced that citation with the existing one (and updated the bibliographic data for the latter—realized that it still had placeholder info).
